KLSadd_insertion/src/updateChapters.py

```diff
@@ -187,162 +411,210 @@ def referencePlacer(chap, references, p, kls, refII, refIII, specref):
-    if chapticker2 == 0:
-        designator = "9."
-    elif chapticker2 == 1:
-        designator = "14."
-    count = 0
-    sectionnuma = specref[0]
-    sectionnumb = specref[1]
-    for i in specref:
-        pass
+# method to change file string(actually a list right now), returns string to be written to file
+# If you write a method that changes something, it is preffered that you call the method in here
+def fix_chapter(chap, references, paragraphs_to_be_added, kls, kls_list_all, chapticker2, new_commands, klsaddparas, sortmatch_2):
+    """
+    Removes specific lines stopping the latex file from converting into python, as well as running the
+    functions responsible for sorting sections and placing the correct additions in the correct places
 
+    :param chap: Chapter being processed.
+    :param references: List containing references (used in reference placer).
+    :param paragraphs_to_be_added: List containing additions to be added.
+    :param kls: The addendum that contains sections to be added (not processed).
+    :param kls_list_all: List of all identified keywords, fed into the chapter sorter.
+    :param chapticker2: Which chapter is being searched (9 or 14).
+    :return:
+    """
 
-# method to change file string(actually a list right now), returns string to be written to file
-# If you write a method that changes something, it is preffered that you
-# call the method in here
-def fixChapter(chap, references, p, kls, refII, refIII, specref):
-    # chap is the file string(actually a list), references is the specific references for the file,
-    # and p is the paras variable(not sure if needed) kls is the KLSadd.tex as
-    # a list
-    referencePlacer(chap, references, p, kls, refII, refIII, specref)
-    chap = prepareForPDF(chap)
-    cms = getCommands(kls)
-    chap = insertCommands(kls, chap, cms)
+    sort_location = 0
+
+    for i in kls_list_all:
+        fix_chapter_sort(kls, chap, i, sort_location, klsaddparas, sortmatch_2)
+        sort_location += 1
+
+    for a in range(len(paragraphs_to_be_added)-1):
+        for b in sortmatch_2:
+            for c in b:
+                with open(DATA_DIR + "compare.tex", "a") as spook:
+                    spook.write(paragraphs_to_be_added[a])
+                    spook.write("NEXT: ")
+                # i = 0
+                # while i < len(c) - 1:
+                #     if c[i:i + 1] == '\n':
+                #         c = c[:i] + c[i + 1:]
+                #     else:
+                #         i += 1
+                # i = 0
+                # shortened_paras = paragraphs_to_be_added[a]
+                # while i < len(shortened_paras) - 1:
+                #     if shortened_paras[i:i + 1] == '\n':
+                #         shortened_paras = shortened_paras[:i] + shortened_paras[i + 1:]
+                #     else:
+                #         i += 1
+                # paragraphs_to_be_added[a] = shortened_paras
+                if "%" == c[-2]:
```

Comment:
Why are these conditions specifically for when the `%` is at the end of the line that you're checking?